test/e2e/e2e_test.go

```diff
@@ -92,6 +106,15 @@ func TestE2E(t *testing.T) {
 	}
 
 	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	// Ensure all pods are running and ready before starting tests (otherwise,
+	// cluster infrastructure pods that are being pulled or started can block
+	// test pods from running, and tests that ensure all pods are running and
+	// ready will fail).
+	if err := waitForPodsRunningReady(api.NamespaceDefault, minStartupPods, podStartupTimeout); err != nil {
```

Comment:
It's a proxy to ensure that we've started starting any addons. Once they're actually being created, they're actually submitted to the system super fast, but ***in theory*** `kube-addons` may not have run yet. In practice, it's basically run immediately after the apiserver comes up.